### PR TITLE
Make file names and exports more consistent

### PR DESCRIPTION
**Pull request by maier49**
**Type:** refactor

**Description:** Makes file and export names, and default exports, more consistent. This doesn't contain any changes in functionality.

**Related Issue:** #???

Please review this checklist before submitting your PR:
- [ ] There is a related issue
- [x] All contributors have signed a CLA
- [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
- [x] The code passes the CI tests
- [ ] Unit or Functional tests are included in the PR
- [x] The PR increases or maintains the overall unit test coverage percentage
- [x] The code is ready to be merged

Use `createX` pattern for all modules that export factory
functions.

Export factory function as default for all modules that
export factory functions.

Move query interface to interfaces file, to separate it
from the createCompoundQuery factory.

### CODE REVIEW

File: src/patch/createJsonPointer.ts [hunk 1]
```diff
@@ -26,7 +25,7 @@ function toString(...segments: string[]): string {
 	});
 }
 
-export function createPointer(...segments: string[]): JsonPointer {
+function createPointer(...segments: string[]): JsonPointer {
```
Comment (kitsonk) [inline src/patch/createJsonPointer.ts]: shouldn't we name this `createJsonPointer` to reflect the modules name?

**Review by kitsonk — COMMENTED** [review]
One comment...

File: src/patch/createJsonPointer.ts (hunk 1 above)
Comment (maier49) [reply src/patch/createJsonPointer.ts]: Yes, fixed.

**Review by kitsonk — APPROVED** [review]
👍 🦃

**Pull request merged**